app/services/admin/api/cms/template_service.rb

```diff
@@ -8,36 +8,27 @@ class UnknownTemplateTypeError < TemplateServiceError; end
   class UnknownSectionError < TemplateServiceError; end
   class UnknownLayoutError < TemplateServiceError; end
 
-
-
-  def initialize(current_account, params)
-    @params = params
+  def initialize(current_account, params, permitted_params)
+    @permitted_params = permitted_params
     @current_account = current_account
+    @type = params.delete('type')&.to_sym
```

Comment:
Why is this moved to the superclass?
Either `type` or `params` don't seem to be used in the Update service...

Reply:
You're right, I'll double check this. The reason is I don't want the `type` param to be sent to the `update` method because that writes the `type` column in the DB. But after separating `permitted_params` from `params` this is not needed.

Reply:
Done